crates/settlement-clients/starknet/src/tests/test.rs

```diff
@@ -0,0 +1,229 @@
+        std::fs::File::open(contract_path.join("mock_contracts_Piltover.contract_class.json"))
+            .expect("Could not open sierra class file"),
+    )
+    .expect("Failed to parse SierraClass");
+
+    let compiled_class: CompiledClass = serde_json::from_reader(
+        std::fs::File::open(contract_path.join("mock_contracts_Piltover.compiled_contract_class.json"))
+            .expect("Could not open compiled class file"),
+    )
+    .expect("Failed to parse CompiledClass");
+
+    let flattened_class = sierra_class.clone().flatten().unwrap();
+    let compiled_class_hash = compiled_class.class_hash().unwrap();
+
+    let DeclareTransactionResult { transaction_hash: declare_tx_hash, class_hash: _ } =
+        account.declare_v2(Arc::new(flattened_class.clone()), compiled_class_hash).send().await.unwrap();
+    println!("declare tx hash {:?}", declare_tx_hash);
+
+    let is_success = wait_for_tx_success(&account, declare_tx_hash, Duration::from_secs(2)).await;
+    assert!(is_success, "Declare trasactiion failed");
+
+    let contract_factory = ContractFactory::new(flattened_class.class_hash(), account.clone());
+    let deploy_v1 = contract_factory.deploy_v1(vec![], felt!("1122"), false);
+    let deployed_address = deploy_v1.deployed_address();
+
+    env::set_var("STARKNET_CAIRO_CORE_CONTRACT_ADDRESS", deployed_address.to_hex_string());
+    let InvokeTransactionResult { transaction_hash: deploy_tx_hash } =
+        deploy_v1.send().await.expect("Unable to deploy contract");
+
+    let is_success = wait_for_tx_success(&account, deploy_tx_hash, Duration::from_secs(2)).await;
+    assert!(is_success, "Deploy trasaction failed");
+
+    let env_settings = EnvSettingsProvider {};
+    let settlement_client = StarknetSettlementClient::new_with_settings(&env_settings).await;
+    let onchain_data_hash = [1; 32];
+    let mut program_output = Vec::with_capacity(32);
+    program_output.fill(onchain_data_hash);
+    let update_state_tx_hash = settlement_client
+        .update_state_calldata(program_output, onchain_data_hash, [1, 1])
+        .await
+        .expect("Sending Update state");
+
+    println!("update state tx hash {:?}", update_state_tx_hash);
+
+    let is_success = wait_for_tx_success(
+        &account,
+        Felt::from_hex(&update_state_tx_hash).expect("Incorrect transaction hash"),
+        Duration::from_secs(2),
+    )
+    .await;
+    assert!(is_success, "Update state transaction failed/reverted");
+
+    let call_result = account
+        .provider()
+        .call(
+            FunctionCall {
+                contract_address: deployed_address,
+                entry_point_selector: selector!("get_is_updated"),
+                calldata: vec![Felt::from_bytes_be_slice(&onchain_data_hash)],
+            },
+            BlockId::Tag(BlockTag::Latest),
+        )
+        .await
+        .expect("failed to call the contract");
+    assert!(call_result[0] == true.into(), "Should be updated");
```

Comment:
doesn't the call return the actual hash?

Reply:
Call does not create a starknet transaction, so it does not have a hash

Reply:
oh right my bad